### PR TITLE
Update OpenJDK latest Docker image to use Java 21

### PR DESCRIPTION
**Pull request by fab-10**
## PR description

Switched base image to `ubuntu:rolling`, that follow any stable Ubuntu release, currently 23.10) since OpenJDK Java 21 is not available on latest LTS Ubuntu.

### CODE REVIEW

File: docker/openjdk-latest/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
 
-FROM ubuntu:22.04
+FROM ubuntu:rolling
 ARG VERSION="dev"
 
 RUN apt-get update && \
- apt-get install --no-install-recommends -q --assume-yes openjdk-19-jre-headless=19* libjemalloc-dev=5.* && \
+ apt-get install --no-install-recommends -q --assume-yes openjdk-21-jre-headless libjemalloc-dev adduser && \
```
Comment (jflo) [inline docker/openjdk-latest/Dockerfile:6]: just curious, what triggered the need for adduser ?

Comment (fab-10) [reply docker/openjdk-latest/Dockerfile:6]: it is no more present in the base image

File: docker/openjdk-latest/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
 
-FROM ubuntu:22.04
+FROM ubuntu:rolling
```
Comment (shemnon) [inline docker/openjdk-latest/Dockerfile:2]: This will make reproducible builds difficult.

Comment (fab-10) [reply docker/openjdk-latest/Dockerfile:2]: what do you suggest to use for the base image?

Comment (fab-10) [reply docker/openjdk-latest/Dockerfile:2]: Fine for me, will set this to `23.10`, was using `rolling` to automatically switch to a supported version, but that could actually breaks the build, like it happened now with the missing `adduser`

Comment (jflo) [reply docker/openjdk-latest/Dockerfile]: I think i can accept that tradeoff for only this build profile.

Comment (fab-10) [reply docker/openjdk-latest/Dockerfile]: @shemnon @jflo are you fine with the current implementation?

**Review by shemnon — APPROVED** [review]

**Review by jflo — APPROVED** [review]

**Pull request merged**